jsapp/js/components/metadataEditor.es6

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import Checkbox from 'js/components/checkbox';
+import TextBox from 'js/components/textBox';
+import {
+  assign,
+  t
+} from 'js/utils';
+import {
+  META_QUESTION_TYPES,
+} from 'js/constants';
+import bem from 'js/bem';
+
+const AUDIT_HELP_URL = 'http://support.kobotoolbox.org/en/articles/2648050-audit-logging-meta-question-type';
+
+/**
+ * @prop {object} survey
+ * @prop {function} onChange
+ */
+export default class MetadataEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      metaProperties: [],
+      auditAppearance: null
+    };
+    autoBind(this);
+  }
+
+  componentDidMount() {
+    this.rebuildState();
+  }
+
+  rebuildState() {
+    const newState = {
+      metaProperties: []
+    };
+    META_QUESTION_TYPES.forEach((metaType) => {
+      const detail = this.getSurveyDetail(metaType);
+      if (detail) {
+        newState.metaProperties.push(assign({}, detail.attributes));
+      }
+    });
+    this.setState(newState);
+  }
+
+  getMetaProperty(metaType) {
+    return this.state.metaProperties.find((metaProp) => {
+      return metaProp.name === metaType;
+    });
+  }
+
+  getSurveyDetail(sdId) {
+    return this.props.survey.surveyDetails.filter((sd) => {
+      return sd.attributes.name === sdId;
+    })[0];
+  }
+
+  onCheckboxChange(name, isChecked) {
+    this.getSurveyDetail(name).set('value', isChecked);
+    this.rebuildState();
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange();
+    }
+  }
+
+  onAuditAppearanceChange(newVal) {
+    this.getSurveyDetail(META_QUESTION_TYPES.get('audit')).set('appearance', newVal);
```

Comment:
The spec actually requires this to be `parameters`, not `appearance` (the original issue description was mistaken). I already have a local commit to fix it.